### DISCUSSION TITLE
Is there a method to get the "raw" data rather than the proxied.

### DISCUSSION THREAD

**Discussion by dinfer** [pmndrs/valtio#956] [Q&A] [answered]
If I want to get data, and avoid tracking, or just want to pass native data(not proxied) to other component, It's more safe to just send pain object rather than proxied values.

for example

```
const globalState = proxy({x: {}})

 // this component may change the properties of `v.x`, and affect the globalState.
<ThirdPartyComponent state={v.x} />
// This is more safe, bug this will mark current component depents all data in the `v.x` object, which I don't want.
<ThirdPartyComponent state={JSON.parse(JSON.stringify(v.x))} /> 
```

Is there any plan to support something like `toRaw` in [reactivity](https://vuejs.org/api/reactivity-advanced.html#toraw) ?

I've searched google and this repo using keyword `toRaw`, there is no result here, so I create a post here, hoping get some help.

**Comment by dai-shi** [comment]
> this component may change the properties of `v.x`

That doesn't sound good. Component function should be pure.

> JSON.parse(JSON.stringify(v.x))

I would do `snapshot(v.x)` which is more efficient, but if you need to modify it, you need to copy it as snapshot is not modifiable.

**Reply by dinfer** [reply]
Thanks for your reply : )
Yes,  they should be pure function, but the code is not always written by you, right? Just want a writable copy to prevent potential non-pure function. 
There are ways to get a copy, for example, using deepClone of lodash, but I don't know how to avoid access recording, the copy may occur in rendering.

**Reply by dai-shi** [reply, marked as answer]
`snapshot(...)` should work, not `useSnapshot(...)`.
